pages/api/longestStreak.ts

```diff
@@ -82,39 +84,50 @@ export function longestSubarray(array: GameBlockState[], x: BlockState) {
   }
   maxlength = Math.max(maxlength, sum);
   sum = 0;
+  if (array[0].state === x && array[array.length-1].state === x) {
+    maxlength +=1
+  }
   return maxlength;
 }
 
-export function calculateWinner(array: GameBlockState[], playerName: string) {
-  let playerOneArray = longestSubarray(
-    array,
-    BlockState.PLAYER_ONE_SELECTED && BlockState.HIGHLIGHTED
-  );
-  console.log("P1", playerOneArray);
+
+
+export function calculateWinner(array: GameBlockState[], functionName: () => void) {
+  let playerOneArray = longestSubarray(array, BlockState.PLAYER_ONE_SELECTED);
   let playerTwoArray = longestSubarray(array, BlockState.PLAYER_TWO_SELECTED);
-  console.log("P2", playerTwoArray);
+  let outcome = functionName()
```

Comment:
This isn't clear what's happening. Can you rename things to be more descriptive or add comments

Reply:
comments added